source/sdk/Character.ooc

```diff
@@ -127,33 +90,30 @@ operator as (value: CString) -> String { value ? value toString() : null }
 
 CString: cover from Char* {
 	clone: func -> This {
-		length := length()
+		length := this length()
 		copy := This new(length)
 		memcpy(copy, this, length + 1)
-		return copy as This
+		copy as This
 	}
 	equals: func ( other: This) -> Bool {
-		if (other == null) return false
-		l := length()
-		l2 := length()
-		if (l != l2) return false
-
-		for (i in 0 .. l) {
-			if (this[i] != other[i]) return false
+		result := false
+		if (other != null) {
+			(length, otherLength) := (this length(), other length())
```

Comment:
Great example of why this is so useful. The original code doesn't even check `other length`, and would eventually crash if it's shorter.